From 209385b01e4452f62c91b279988a1c877030e8b7 Mon Sep 17 00:00:00 2001
From: Marius Meschter <marius.meschter@univention.de>
Date: Wed, 13 Aug 2025 13:07:11 +0200
Subject: [PATCH 2/2] fix(udl): fix module initialization

fix the listener-module initialization if the return value of the LDAP
search is LDAP_NO_SUCH_OBJECT on the last iteration of the loop.

Issue univention/dev/internal/team-nubus#1372
---
 .../univention-directory-listener/src/change.c    | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/management/univention-directory-listener/src/change.c b/management/univention-directory-listener/src/change.c
index 08a43dea1d..0ccc89b0eb 100644
--- a/management/univention-directory-listener/src/change.c
+++ b/management/univention-directory-listener/src/change.c
@@ -163,10 +163,19 @@ static int change_init_module(univention_ldap_parameters_t *lp, Handler *handler
 					cache_free_entry(NULL, &cache_entry);
 					abort_init = true;
 					goto cleanup;
+				/*
+				Ignore LDAP_NO_SUCH_OBJECT. An object can be
+				deleted after we do the ldapsearch. We
+				shouldn't need to care here.
+
+				However we should reset the return value to 0
+				Otherwise, if LDAP_NO_SUCH_OBJECT occurs in the last iteration of this loop
+				it becomes the return value of the function.
+				*/
+				} else if (rv == LDAP_NO_SUCH_OBJECT) {
+					univention_debug(UV_DEBUG_LISTENER, UV_DEBUG_ALL, "could not find DN %s", dns[i].dn);
+					rv = LDAP_SUCCESS;
 				}
-				/* Ignore LDAP_NO_SUCH_OBJECT. An object can be
-				   deleted after we do the ldapsearch. We
-				   shouldn't need to care here. */
 			} else if (rv != 0) {
 				univention_debug(UV_DEBUG_LISTENER, UV_DEBUG_WARN, "error while reading from database");
 				rv = LDAP_OTHER;
-- 
2.50.1

